File: 1556-make-two-arrays-equal-by-reversing-subarrays/make-two-arrays-equal-by-reversing-subarrays.py
class Solution(object):
    def canBeEqual(self, target, arr):
        """
        :type target: List[int]
        :type arr: List[int]
        :rtype: bool
        """
        n=len(target)
        ara=[0]*(1003)
        for i in range(n):
            ara[target[i]]+=1
            ara[arr[i]]-=1

        for i in range(1,1001):
            if ara[i]!=0:
                return False
        
        return True;

File: 1556-make-two-arrays-equal-by-reversing-subarrays/make-two-arrays-equal-by-reversing-subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        int n=target.size();
        int ara[1003];
        memset(ara,0,sizeof(ara));
        for(int i=0;i<n;i++){
            ara[target[i]]++;
            ara[arr[i]]--;
        }
        for(int i=1;i<=1000;i++){
            if(ara[i]!=0){
                return false;
            }
        }
        return true;
    }
};